### PR TITLE
[uma/lma] Add BGP confederation route-propagation tests

### PR DESCRIPTION
**Pull request by wenyiz2021**
This pull request adds a comprehensive set of BGP confederation route-propagation tests for `lma` (LowerMgmtAggregator) and `uma` (UpperMgmtAggregator) devices. The tests verify correct AS presentation, AS-path integrity, and cross-boundary route propagation within a BGP confederation, ensuring the devices behave according to BGP confederation standards. The tests are designed to be safe to run from any control host and do not require neighbor VM access.

**Test Coverage for Key Confederation Behaviors**
* Implemented tests to verify that the DUT presents its member AS to confederation-internal peers and the confederation identifier to external peers (`test_confed_local_as_presentation`).
* Added checks to ensure member sub-AS numbers only appear inside AS_CONFED_SEQUENCE segments and that the confederation identifier does not leak into the local RIB (`test_confed_as_path_integrity`).
* Verified that the default route originated on one side of the confederation boundary is learned and re-advertised

```
LMA DUT (-10) — member 64589, confed id 64582, confed peers {64588}:
AS presentation: external peers (AS 65300, M2/M3) see localAs=64582; confed-internal peers (AS 64588, UMA) see localAs=64589.
Cross-boundary propagation: default learned from confed-internal UMA → re-advertised to external M2/M3.
IPv4: origin 10.0.0.1, 10.0.0.3 → advertised to 10.0.0.137/139/141/143
IPv6: origin fc00::2, fc00::6 → advertised to fc00::112/116/11a/11e

UMA DUT (-9) — member 64588, confed id 64582, confed peers {64589}:
AS presentation: external peers (RWA AS 65200, M1 AS 65300) see localAs=64582; confed-internal peers (AS 64589, LMA) see localAs=64588.
Cross-boundary propagation (mirror direction): default learned from external RWA → re-advertised to confed-internal LMA.
IPv4: origin 10.0.0.1 → advertised to 10.0.0.137/139/141/143
IPv6: origin fc00::2 → advertised to fc00::112/116/11a/11e
```

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] New Test case
    - [ ] Skipped for non-supported platforms
- [ ] Test case improvement

### Back port request

- [ ] 202311
- [ ] 202405
- [ ] 202411
- [ ] 202505
- [ ] 202511
- [ ] 202512
- [ ] 202605

Tracking issue/work item for backport/cherry-pick request (GitHub issue or Microsoft ADO):
Failure type: 

### Tested branch

- [ ] master
- [ ] 202311
- [ ] 202405
- [ ] 202411
- [ ] 202505
- [ ] 202511
- [ ] 202512
- [ ] 202605
- [ ] N/A

#### How did you verify/test it?
Tested on LMA and UMA testbeds internally

-10 uma topology:
https://elastictest.org/scheduler/testplan/6a767ddd32974924d6a7e53a?testcase=bgp%2Ftest_bgp_confed_route_propagation.py&type=console
-9 lma topology:
https://elastictest.org/scheduler/testplan/6a7a0eb66fff0a820f1b3218

### CODE REVIEW

File: tests/bgp/test_bgp_confed_route_propagation.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('lma', 'uma'),
```
Comment (bingwang-ms) [inline tests/bgp/test_bgp_confed_route_propagation.py:42]: Can we also put `lt2`, `ft2` and `ut2` in the supported topo?

**Comment by bingwang-ms** [comment]
> _Review posted by Steve, an AI assistant, on behalf of @bingwang-ms._

Thanks for the tests — the module is well-scoped and the docstrings actually explain the "why", which is unusually nice for BGP corner-case coverage. A few things below; none of the code notes are blockers, but the PR metadata one is worth addressing before merge.

### PR metadata (please address)

The `Request for msft-202603 branch` label is set, but the template sections are empty:

- **Type of change** — nothing checked.
- **Back port request** — no target branch checked, no linked GitHub issue / ADO work item. The template explicitly says backport requests without a tracker "may not be favored."
- **Tested branch** / **Test result** — nothing checked; the elastictest links only appear under "How did you verify".

Could you either fill these in (branch, tracker, per-branch test evidence) or drop the msft-202603 label if the backport isn't actually being requested?

### Topology marker

`pytest.mark.topology('lma', 'uma')` looks correct as-is. LT2/FT2 topos in `ansible/vars/` use plain private ASNs and have no `dut_confed_asn` / `dut_confed_peers`, and there is no `topo_ut2.yml` today. The tests would fail hard in `_get_confed_config` on non-confed topos (the `bgp confederation identifier` regex asserts). Two small suggestions:

1. Consider making `_get_confed_config` `pytest.skip(...)` (rather than `pytest_assert`) when confederation config is absent — belt-and-suspenders if someone runs it on a non-confed testbed by mistake.
2. If there is a future UT2 (or other confed-capable topo) planned, worth noting in the docstring so the marker can be extended later without re-reading the module.

### Code comments (suggestions, not blockers)

1. **`_get_confed_config` assumes a single default-VRF BGP instance.**
   `re.search(r'^router bgp (\d+)', ..., re.MULTILINE)` returns the first match, and the `bgp confederation identifier` / `bgp confederation peers` regexes are not anchored to that block. On a config with additional `router bgp N vrf X` sections this could pick up mismatched fields. For lma/uma today this is fine — I'd just add a comment stating the assumption, or restrict the parse to the default-VRF block.

2. **`test_confed_default_route_crosses_boundary` direction selection.**
   ```python
   origin_internal = any(ip in internal for ip in origin_peer_ips)
   opposite_side = external if origin_internal else internal
   ```
   If the default is learned from *both* sides simultaneously, only the external re-advertisement is verified. The docstring says "re-advertised to at least one peer on the opposite side of that boundary", which is looser than the code's single-direction check. Either tighten the docstring to match the intended one-directional model, or check both directions when both sides originate.

3. **Nit — inline conditional command string.**
   ```python
   default_info = _vtysh_json(duthost, "show ip bgp {} json".format(prefix) if ip_version == 4
                              else "show bgp ipv6 {} json".format(prefix))
   ```
   Slightly awkward compared to the `cmd = ...` pattern used in `_get_bgp_routes` / `_get_advertised_prefixes`. Hoisting to a local would keep the style consistent.

4. **Nit — `_split_confed_and_normal_asns`** only handles `AS_CONFED_SEQUENCE` (`(...)`), not `AS_CONFED_SET` (`[...]`). Matches your docstring, so purely a heads-up if FRR ever emits sets on these topos.

Once the metadata is sorted (and ideally item 1 above gets a one-liner comment), the code side looks good to me.

**Review by yejianquan — APPROVED** [review]
LGTM

File: tests/bgp/test_bgp_confed_route_propagation.py (hunk 1 above)
Comment (yejianquan) [reply tests/bgp/test_bgp_confed_route_propagation.py:42]: I'm going to merge to unblock DMA qualification, @wenyiz2021 @bingwang-ms we may creating another PR to follow on enabling on DT2

Comment (wenyiz2021) [reply tests/bgp/test_bgp_confed_route_propagation.py:42]: hi @bingwang-ms  I think it should also work on ut2/lt2 but I didnt qualify yet since they have many different platform and sku. can enable them in another item with fully qualifications
